modules/hal_silabs/simplicity_sdk/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+    ${RADIO_DIR}/rail_lib/chip/efr32/efr32xg2x
+  )
+
+  zephyr_include_directories(
+    ${RADIO_DIR}/rail_lib/common
+    ${RADIO_DIR}/rail_lib/plugin/pa-conversions
+    ${BLUETOOTH_DIR}/bgstack/ll/inc
+  )
+
+  # sl_protocol_crypto
+  zephyr_sources_ifdef(CONFIG_BT_SILABS_HCI
+    ${SECURITY_DIR}/sl_component/sl_protocol_crypto/src/sli_radioaes_management.c
+    ${SECURITY_DIR}/sl_component/sl_protocol_crypto/src/sli_protocol_crypto_radioaes.c
+  )
+
+  if(CONFIG_BT_SILABS_HCI)
+    # prebuilt libs
+    add_prebuilt_library(liblinklayer protocol/bluetooth/bgstack/ll/lib/libbluetooth_controller_efr32xg${SILABS_DEVICE_FAMILY_NUMBER}_gcc_release.a)
+    add_prebuilt_library(libbgcommon  protocol/bluetooth/bgcommon/lib/build/gcc/cortex-m33/bgcommon/release/libbgcommon.a)
+
+    # link mbedTLS
+    if(CONFIG_MBEDTLS)
+      zephyr_link_libraries(mbedTLS)
+    endif()
+  endif()
+
+  if(CONFIG_SOC_GECKO_USE_RAIL)
+    # rail
+    zephyr_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_curves_efr32.c)
+    zephyr_sources(${RADIO_DIR}/rail_lib/plugin/pa-conversions/pa_conversions_efr32.c)
+
+    # prebuilt libs
+    add_prebuilt_library(librail platform/radio/rail_lib/autogen/librail_release/librail_efr32xg${SILABS_DEVICE_FAMILY_NUMBER}_gcc_release.a)
+
+    zephyr_include_directories_ifdef(CONFIG_SOC_GECKO_CUSTOM_RADIO_PHY
+      ${RADIO_DIR}/rail_lib/protocol/ble
+      ${RADIO_DIR}/rail_lib/protocol/ieee802154
+      ${RADIO_DIR}/rail_lib/protocol/zwave
+      ${RADIO_DIR}/rail_lib/protocol/sidewalk
+      ${RADIO_DIR}/rail_lib/plugin/rail_util_protocol
+      ${RADIO_DIR}/rail_lib/plugin/rail_util_protocol/config/efr32xg${SILABS_DEVICE_FAMILY_NUMBER}
+    )
+    zephyr_sources_ifdef(CONFIG_SOC_GECKO_CUSTOM_RADIO_PHY
+      ${RADIO_DIR}/rail_lib/plugin/rail_util_protocol/sl_rail_util_protocol.c
+    )
+  endif()
+endif()
+
+zephyr_include_directories(
+  ${DEVICE_DIR}/SiliconLabs/${SILABS_DEVICE_FAMILY}/Include
+  ${COMMON_DIR}/config
+  ${COMMON_DIR}/inc
+  ${EMLIB_DIR}/inc
+  ${PERIPHERAL_DIR}/inc
+  ${SERVICE_DIR}/device_init/config/s2
+  ${SERVICE_DIR}/device_init/config/s2/sdid${CONFIG_SOC_GECKO_SDID}
+  ${SERVICE_DIR}/clock_manager/inc
+  ${SERVICE_DIR}/device_init/inc
+  ${SERVICE_DIR}/device_manager/inc
+  ${SERVICE_DIR}/hfxo_manager/config
+  ${SERVICE_DIR}/hfxo_manager/inc
+  ${SERVICE_DIR}/hfxo_manager/src
+  ${SERVICE_DIR}/memory_manager/inc
+  ${SERVICE_DIR}/memory_manager/profiler/inc
+  ${SERVICE_DIR}/power_manager/config
+  ${SERVICE_DIR}/power_manager/inc
+  ${SERVICE_DIR}/power_manager/src
+  ${SERVICE_DIR}/sleeptimer/config
+  ${SERVICE_DIR}/sleeptimer/inc
+  ${SERVICE_DIR}/sleeptimer/src
+  ${SECURITY_DIR}/sl_component/sl_protocol_crypto/src
+  ${BOARD_DIR}
+)
+
+zephyr_compile_definitions(
+  ${SILABS_DEVICE_PART_NUMBER}
+)
+
+zephyr_compile_definitions_ifdef(CONFIG_SOC_GECKO_DEV_INIT
+  SL_CATALOG_POWER_MANAGER_PRESENT
+  SL_CATALOG_HFXO_MANAGER_PRESENT
+)
+
+zephyr_compile_options(
```

Comment:
zephyr_library_...? (same for others)

Reply:
I can change that, though most of this is just copy-paste from the CMake files as they were in the HAL tree.